src/parser/chunkparser.hpp

```diff
@@ -0,0 +1,61 @@
+#pragma once
+#include <array>
+#include <nlohmann/json.hpp>
+
+namespace docgen {
+
+//////////////////////////////////////////////////////
+// ChunkParser Declaration
+//////////////////////////////////////////////////////
+
+// ChunkParser is a class that will parse a chunk of text into JSON format.
+class ChunkParser
+{
+public:
+
+    // Main function that user will call to parse a chunk of text.
+    // This function will update the json object with information extracted from chunk.
+    void parse(const char *begin, 
+               const char *end);
+
```

Comment:
I think we need a function to retrieve the json.

Reply:
ah right my b!

Reply:
I am only putting a const version since we don't want users to directly modify this private member. This is sufficient for our use-case also since we're just gonna use this reference to copy-construct.

BTW, we can technically bypass the need for copy-constructing this json later...
Have a vector of these json objects default-constructed, and simply pass by reference each element into `parse` function as the third parameter... Now I'm leaning more towards this approach...